src/main/java/org/zalando/nakadi/service/EventStreamConfig.java

```diff
@@ -150,7 +157,7 @@ public Builder withBatchTimeout(@Nullable final Integer batchTimeout) {
         }
 
         public Builder withStreamTimeout(@Nullable final Integer streamTimeout) {
-            if (streamTimeout != null) {
+            if (streamTimeout != null && streamTimeout <= MAX_STREAM_TIMEOUT && streamTimeout > 0) {
```

Comment:
Do you think the users use case will be satisfied with that constraint? E.g. they require 1.5 hours to process some queries. I mean leave it for users, if they provide it they need it

Reply:
Well, it will be strange if we define the max. limit but allow users to use value greater than this limit. Also if we allow that - that can kill the whole idea of connections shuffling (e.g. if couple of heavy subscriptions will set limit to 1 week)

Reply:
Then do not define max value, define only default one. I do not think that there are such limits, mostly people use default, which is infinity, which causing the issues.

Reply:
Ok, I see your point. But I actually wanted to limit the max value not only introduce the default one. If we leave the possibility to create super long living connections then we might still have the problem we try to solve. But let's discuss this point tomorrow after standup to get opinion of other team members.